### PR TITLE
Follow NavLinks that do not match a registered route

### PR DESCRIPTION
**Pull request by mridgway**
Extension of #369.

This renames `autoMatch` to `validate` and removes the `makePath` execution on the `href` prop.

### CODE REVIEW

**Comment by yahoocla** [comment]
Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! :smile:

**Comment by yahoocla** [comment]
CLA is valid!

File: packages/fluxible-router/docs/api/NavLink.md [hunk 1]
```diff
@@ -14,7 +14,12 @@ in navigation events.
 | replaceState | boolean, default to false | If set to true, replaceState is being used instead of pushState |
 | preserveScrollPosition | boolean, default to false | If set to true, the page will maintain its scroll position while change to new route.  This is useful for cases like on e-commerce site where user might want to change a product size.  The url will change, but the scroll position needs to remain the same. |
 | stopPropagation | boolean, default to false | If set to true, the click event will not be propagated beyond the NavLink |
+| activeClass | string | Class to add to `className` when link is active |
+| activeStyle | object | Inline styles to merge with `style` when link is active |
 | activeElement | string or React.Component| If specified, active NavLink components will use this component instead of 'a' tags |
+| validate | boolean, default to false | Useful when a it is unknown whether a link matches a registered route, this
```
Comment (redonkulus) [inline packages/fluxible-router/docs/api/NavLink.md]: Useful when it is ... (remove the "a")

**Comment by redonkulus** [comment]
One minor issue, :+1:

File: packages/fluxible-router/lib/createNavLinkComponent.js [hunk 2]
```diff
@@ -132,7 +134,7 @@ module.exports = function createNavLinkComponent (overwriteSpec) {
                 return;
             }
 
-            if (href[0] !== '/') {
+            if (href[0] !== '/' || (this.props.validate && !routeStore.getRoute(href))) {
```
Comment (lingyan) [inline packages/fluxible-router/lib/createNavLinkComponent.js]: Should we move `(this.props.validate && !routeStore.getRoute(href))` check to line 150 to be:
```
if (this.props.validate && !routeStore.getRoute(href)) {
    return;
}
```

Line 148 will make sure absolute urls with the same origin with be converted to relative url, which makes it perfect for matching route.

Comment (mridgway) [inline packages/fluxible-router/lib/createNavLinkComponent.js]: I like it. I'll make the change.

**Comment by mridgway** [comment]
Updated.

**Comment by lingyan** [comment]
Perfect. :+1:

**Pull request merged**